platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java

```diff
@@ -477,22 +479,61 @@ protected LDAPConnectionFactory createLdapConnectionFactory(String url, Boolean
         return new LDAPConnectionFactory(host, port, lo);
     }
 
-    /**
-     * Added additional logging to the security LOGGER.
-     */
-    @Override
-    public boolean login() throws LoginException {
-        try {
-            boolean isLoggedIn = doLogin();
-
-            if (!isLoggedIn) {
-                SecurityLogger.audit("Username [" + user + "] failed LDAP authentication.");
+    void validateUsername(String username) throws LoginException {
+        boolean hasBadCharacters = false;
+        for (int i = 0; i < username.length(); i++) {
+            char curChar = username.charAt(i);
+            switch (curChar) {
+            case '\\':
```

Comment:
Hmm. We should probably ask Scott. It's his code. I read somewhere that this is fastest way to do it, but I don't know if that was the reason. I'll tag @stustison in the comment to give him a chance to chime in.